lms/lmsweb/views.py

```diff
@@ -66,22 +66,35 @@ def load_user(user_id):
 
 
 @webapp.route('/login', methods=['GET', 'POST'])
-def login():
+def login(login_error=None):
+
     if current_user.is_authenticated:
         return get_next_url(request.args.get('next'))
 
     username = request.form.get('username')
     password = request.form.get('password')
     next_page = request.form.get('next')
+    login_error = request.args.get('login_error')
     user = User.get_or_none(username=username)
 
-    if user is not None and user.is_password_valid(password):
-        login_user(user)
-        return get_next_url(next_page)
-    elif user is not None:
-        return redirect(url_for('login', **{'next': next_page}))
-
-    return render_template('login.html')
+    if request.method == 'POST':
+        if user is not None and user.is_password_valid(password):
+            login_user(user)
+            return get_next_url(next_page)
+        elif user is not None and user.is_password_valid(password) is False:
+            login_error = 'Wrong password'
+            return redirect(
+                url_for('login', **{'next': next_page,
+                        'login_error': login_error}),
+            )
+        elif user is None:
+            login_error = 'Username does not exist, please register first'
```

Comment:
We shouldn't have a different message for user that doesn't exist, that's might lead to [username enumeration](https://owasp.org/www-project-web-security-testing-guide/latest/4-Web_Application_Security_Testing/03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account).

Reply:
wasn't aware of this kind of hacking, was an interesting read! changed the error message